File: Labs/Lab04 JAVA/TechLead.java
public class TechLead extends Funcionario implements InterfaceGerente, InterfaceDesenvolvedor {
    private int numeroDeEquipes;
    private String linguagemPrincipal;
    private int projetosEmAndamento;

    public TechLead(String nome, double salarioBase, int numeroDeEquipes, String linguagemPrincipal,int projetosEmAndamento) {
        super(nome, salarioBase);
        this.numeroDeEquipes = numeroDeEquipes;
        this.linguagemPrincipal = linguagemPrincipal;
        this.projetosEmAndamento = projetosEmAndamento;
    }

    @Override
    public int getNumeroDeEquipes() {
        return numeroDeEquipes;
    }

    @Override
    public double calcularBonus() {
        return numeroDeEquipes * 500;
    }

    @Override
    public String getLinguagemPrincipal() {
        return linguagemPrincipal;
    }

    @Override
    public String projetosRealizados() {
        return "Projetos entregues na linguagem " + linguagemPrincipal;
    }

    @Override
     public String descricao() {
        return super.descricao() + ", Número de equipes: " + numeroDeEquipes +
               ", Linguagem principal: " + linguagemPrincipal +
               ", Projetos em andamento: " + projetosEmAndamento +
               ", Bônus: " + calcularBonus();
    }
}

File: Lab04 C++/TechLead.cpp
#include "TechLead.hpp"

using namespace std;

TechLead::TechLead(const string &nome, int salario_base, int numero_equipes, const string &linguagem_principal, int projetos_em_andamento)
    : Funcionario(nome, salario_base),
      Gerente(nome,salario_base,numero_de_equipes),
      Desenvolvedor(nome,salario_base,linguagem_principal),
      projetos_em_andamento(projetos_em_andamento) {}

int TechLead::getProjetosEmAndamento()
{
    return projetos_em_andamento;
}

string TechLead::descricao() 
{
    return Funcionario::descricao() + ", Número de equipes: " + to_string(getNumero_de_equipes()) +
           ", Bônus: " + to_string(bonus()) + ", Linguagem principal: " + getLinguagemPrincipal() +
           ", Projetos em andamento: " + to_string(projetos_em_andamento);
}